.github/workflows/release.yaml

```diff
@@ -30,14 +32,50 @@ jobs:
         run: make GIT_TAG=${{ github.event.inputs.tag }} -f builder.Makefile cross
 
       - name: Compute checksums
-        run: cd bin; for f in *; do shasum --binary --algorithm 256 $f | tee -a checksums.txt > $f.sha256; done
+        working-directory: bin/
+        run: for f in *; do shasum --binary --algorithm 256 $f | tee -a checksums.txt > $f.sha256; done
+
+      - name: Generate SLSA subjects
+        id: hash
+        working-directory: bin/
+        run: |
+          set -euo pipefail
+          echo "::set-output name=hashes::$(cat checksums.txt | base64 -w0)"
 
       - name: License
         run: cp packaging/* bin/
 
+      - uses: actions/upload-artifact@v3
+        with:
+          name: "bin"
+          path: "bin/*"
+
+  provenance:
+    needs: [build]
+    permissions:
+      actions:  read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.0
+    with:
+      base64-subjects: "${{ needs.build.outputs.hashes }}"
+
+  release:
+    needs: [build, provenance]
+    runs-on: ubuntu-latest
+    steps:
+      # Download binaries and license.
+      - uses: actions/download-artifact@v3
+        with:
+          name: "bin"
+      # Download provenance file.
+      - uses: actions/download-artifact@v3
+        with:
+          name: "${{ needs.provenance.outputs.attestation-name }}"
+      # Upload to release.
```

Comment:
```suggestion
      # Verify binaries if dry run
      - name: Verify binaries
        if: 'inputs.dry-run' 
        run: |
          echo TODO: verify with slsa-verifier
      # Upload to release.
```

Reply:
I guess we need to put in the actual steps here. Don't suppose you have a packaged action for verifying yet?

Reply:
Not yet, but we are working on it, coming in the next release. 
Would you like me to add the steps in this PR or do that in a follow-up PR?
We have 2 options:
- `go install ...@vx.y`. This will build from source. Dependabot and removatebot don't parse commands, so you won't receive updates. This may be acceptable temporarily.
- pull the binary from our release. Can verify its hash as well, which are committed to the repo and reviews are enforced https://github.com/slsa-framework/slsa-verifier/blob/main/SHA256SUM.md. The verification will part of a different job (after the generation), and we can make this job have just `permissions: none` and not worry about verifying hashes - since a compromise would not affect the workflow / repository with no permissions

Let me know what you prefer.

Reply:
The cheapest option is to use the pre-compiled binary. But if you don't mind 2-3mn to build the verifier, it's the simplest option.

Reply:
I added the code for verification. PTAL

I also created a tracking issue https://github.com/slsa-framework/slsa-verifier/issues/206 on our repo as an AI to send a PR to update your workflow when we have the GHA released.